### PR TITLE
SQL-883: Adding ClientInfo property to include client name/version to…

### PR DESCRIPTION
**Pull request by bucaojit**
… driver info

`ClientInfo` is an optional property that users can add a client name and version to the mongoDriverInformation, which is used to provide additional client metadata for connections.
Format is:
property["ClientInfo"]="name|version";

The information logged for "tableau-connector|f8807d3" will look like:
```
"clientMetadata": {"driver":{"name":"mongo-java-driver|sync|mongodb-jdbc|tableau-connector","version":"3.12.11|2.0.0-beta-2-6-gc915429|f8807d3"}
```

### CODE REVIEW

File: src/main/java/com/mongodb/jdbc/MongoConnection.java
Comment (nbagnard) [inline src/main/java/com/mongodb/jdbc/MongoConnection.java:74]: If we keep adding more and more connection properties to MongoConnection maybe we'll want to have a dedicated data structure for it. It seems that this is still Ok right now, but if you think it makes sense to switch to a data structure (a simple POJO will do), feel free to do it.

**Review by nbagnard — CHANGES_REQUESTED** [review]

**Review by mattChiaravalloti — CHANGES_REQUESTED** [review]
agreed with Natacha's suggestion regarding refactoring the builder code. I'll check back when that's addressed!

File: src/main/java/com/mongodb/jdbc/MongoConnection.java
Comment (bucaojit) [reply src/main/java/com/mongodb/jdbc/MongoConnection.java]: I made a change to add the driver name to appName and append `|<clientInfo>` if it exists.  
Example log line looks like:
```
2022-05-16T19:01:05.310-0700    ^[[34m[INFO]^[[0m       frontend.command        command execution complete      {"version": "built-without-version-string", "gitVersion": "built-without-git-spec", "pid": 10152, "connectionID": 209902546, "remoteIP": "127.0.0.1", "hostname": "localhost", "tenantID": "5ecd3e8aa1b9b07b3ddd1b4b", "dataLakeType": "UNKNOWN", "correlationID": "16efc1f96e543ff09579239d", "x-request-id": "16efc1f96e543ff09579239d", "commandName": "isMaster", "appName": "mongodb-jdbc|tableau-connector", "elapsed": "7µs", "elapsedNanos": 7000}
```
The way the code change is, if the user adds `appName` to their connection string then that will be used instead.

It's now in both appName and MongoDriverInformation.

File: src/main/java/com/mongodb/jdbc/MongoConnection.java
Comment (bucaojit) [reply src/main/java/com/mongodb/jdbc/MongoConnection.java]: Made the first suggestion to keep the ordering of driver followed by client info.  Let me know if it is as expected.

File: src/main/java/com/mongodb/jdbc/MongoConnection.java
Comment (bucaojit) [reply src/main/java/com/mongodb/jdbc/MongoConnection.java]: I added a POJO MongoConnectionProperties to address this.

File: src/main/java/com/mongodb/jdbc/MongoConnection.java
Comment (nbagnard) [inline src/main/java/com/mongodb/jdbc/MongoConnection.java:109]: The driver version and connector version will not be in `appName`. Is this intentional?

Comment (nbagnard) [reply src/main/java/com/mongodb/jdbc/MongoConnection.java:109]: Also, just saw this while connecting with mongosh : appName=mongosh+1.2.2
What about we follow the convention and use something similar like `driverName+driverVersion|tableau+connectorVersion`?

Comment (bucaojit) [reply src/main/java/com/mongodb/jdbc/MongoConnection.java:109]: Sounds good, I'll add the version to the appName.

File: src/main/java/com/mongodb/jdbc/MongoConnection.java
Comment (bucaojit) [reply src/main/java/com/mongodb/jdbc/MongoConnection.java]: Made the change to store `connectionProperties.getClientInfo()` in a variable here.

File: src/main/java/com/mongodb/jdbc/MongoConnectionProperties.java
Comment (bucaojit) [reply src/main/java/com/mongodb/jdbc/MongoConnectionProperties.java]: It was to match the other POJOs such as `JsonSchema` which has public variables.  I have changed them to `protected` and added setters.

File: src/main/java/com/mongodb/jdbc/MongoConnection.java
Comment (bucaojit) [reply src/main/java/com/mongodb/jdbc/MongoConnection.java:109]: Here's an example output: `"appName": "mongodb-jdbc+2.0.0-beta-2-7-g1c59d6b-dirty|tableau-connector+f8807d3"`

File: src/main/java/com/mongodb/jdbc/MongoConnectionProperties.java
Comment (nbagnard) [reply src/main/java/com/mongodb/jdbc/MongoConnectionProperties.java]: You don't need getters and setters if the fields are protected because you used them in the same package.
However, I think making them private and only allow setting the value via the constructor makes sense because all the info should be known at the time of creation and not modified.

**Review by nbagnard — APPROVED** [review]
Only a few things that you could polish out, but I leave it to you and Matt to decide if these are blockers.
The logic itself looks good to me 👍

File: src/main/java/com/mongodb/jdbc/MongoDriver.java
Comment (mattChiaravalloti) [inline src/main/java/com/mongodb/jdbc/MongoDriver.java:62]:
```suggestion
    public static final String CLIENT_INFO = "clientInfo";
```
[nit] the other options start with lower case letters -- should we follow that pattern here as well?

**Review by mattChiaravalloti — APPROVED** [review]
Looks good! I added one last small request that you might want to consider before merging 👍

File: src/main/java/com/mongodb/jdbc/MongoDriver.java
Comment (bucaojit) [reply src/main/java/com/mongodb/jdbc/MongoDriver.java:62]: Sounds good, I'll make the change.

**Comment by bucaojit** [comment]
evergreen merge

**Pull request merged**
